packages/graphql_server/src/scraping/twitterScraping.ts

```diff
@@ -0,0 +1,111 @@
+import axios, { AxiosRequestConfig } from 'axios'
+import { TwitterPost, TwitterUser, TwitterProfileResponse } from '../../types/twitterScraping'
+
+const username = process.env.SCRAPING_BOT_USER_NAME || ''
+const apiKey = process.env.SCRAPING_BOT_API_KEY || ''
+const apiEndPoint = process.env.SCRAPING_BOT_API_ENDPOINT || ''
+const auth = 'Basic ' + Buffer.from(username + ':' + apiKey).toString('base64')
+
+const SORT_BY_RETWEETS = 'retweets'
+const SORT_BY_LIKES = 'likes'
+
+function sleep(ms: number): Promise<void> {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms)
+  })
+}
+
+/**
+ * Fetches the Twitter user by his handle
+ * @param twitterHandle
+ * @returns the TwitterUser with id, name and count of followers
+ */
+async function getTwitterUserByHandle(twitterHandle: string): Promise<TwitterUser> {
+  try {
+    // builds the axios request configuration
+    const requestConfig: AxiosRequestConfig = {
+      headers: {
+        Accept: 'application/json',
+        Authorization: auth
+      }
+    }
+    // specifies the used scraping mechanism, which is in this case twitterProfile
+    const requestData = {
+      scraper: 'twitterProfile',
+      url: 'https://twitter.com/' + twitterHandle
+    }
+
+    const response = await axios.post<{ responseId: string }>(
+      apiEndPoint,
+      requestData,
+      requestConfig
+    )
+
+    let data: TwitterProfileResponse | null = null
+
+    // sends the request until finalData is received
+    do {
+      await sleep(5000)
```

Comment:
Add comment explaining why we have to wait 5s here for future reference

Reply:
this is weird, is there no immediate request/response callback from scraping-bot? Should we provide an endpoint that it will call when it finished scraping?